torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -95,11 +155,11 @@ def write_single_view(prefix: str, tensor: Tensor, base_index: int):
                 and len(base.size()) == len(tensor.size())
                 and base.size() == tensor.size()
             ):
-                kwargs[f"{prefix}_alias"] = True
+                use_alias()
+            elif (slice_info := try_use_slice(base, tensor)) is not None:
+                use_slice(*slice_info)
             else:
-                kwargs[f"{prefix}_size"] = tensor.size()
-                kwargs[f"{prefix}_stride"] = tensor.stride()
-                kwargs[f"{prefix}_storage_offset"] = tensor.storage_offset()
+                use_as_strided(tensor)
```

Comment:
is there info somewhere on why `as_strided` is worse for us than `slice()` (for inductor?)

Reply:
so there is no benchmark to show, but talking to @eellison in lowering code, you can we realize(), which we do not on the slice path.